pkg/cmd/cluster/connect.go

```diff
@@ -445,13 +445,14 @@ func getUserAndPassword(clusterDef *appsv1alpha1.ClusterDefinition, compDef *app
 	}
 	// 0.8 API connect
 	if secret == nil && compDef != nil {
-		for i := range compDef.Spec.SystemAccounts {
-			if compDef.Spec.SystemAccounts[i].InitAccount {
-				for j, s := range secrets.Items {
-					if strings.Contains(s.Name, compDef.Spec.SystemAccounts[j].Name) {
-						secret = &secrets.Items[j]
-						break
-					}
+		for _, account := range compDef.Spec.SystemAccounts {
+			if !account.InitAccount {
+				continue
+			}
+			for i, s := range secrets.Items {
+				if s.Name == fmt.Sprintf("%s-%s-account-%s", cluster, compDef.Name, account.Name) {
+					secret = &secrets.Items[i]
```

Comment:
the second filed is not the name of ComponentDefinition, but the name of Component